src/sagemaker/cli/compatibility/v2/modifiers/renamed_params.py

```diff
@@ -209,3 +216,93 @@ def old_param_name(self):
     def new_param_name(self):
         """The new name for the image URI argument."""
         return "image_uri"
+
+
+class EstimatorCreateModelImageURIRenamer(MethodParamRenamer):
+    """A class to rename ``image`` to ``image_uri`` in estimator ``create_model()`` methods."""
+
+    @property
+    def calls_to_modify(self):
+        """A mapping of ``create_model`` to common variable names for estimators."""
+        return {
+            "create_model": (
+                "estimator",
+                "chainer",
+                "mxnet",
+                "mx",
+                "pytorch",
+                "rl",
+                "sklearn",
+                "tensorflow",
+                "tf",
+                "xgboost",
+                "xgb",
+            )
+        }
+
+    @property
+    def old_param_name(self):
+        """The previous name for the image URI argument."""
+        return "image"
+
+    @property
+    def new_param_name(self):
+        """The new name for the the image URI argument."""
+        return "image_uri"
+
+
+class SessionCreateModelImageURIRenamer(MethodParamRenamer):
+    """A class to rename ``primary_container_image`` to ``image_uri``.
+
+    This looks for the following calls:
+
+    - ``sagemaker_session.create_model_from_job()``
+    - ``sess.create_model_from_job()``
+    """
+
+    @property
+    def calls_to_modify(self):
+        """A mapping of ``create_model_from_job`` to common variable names for Session."""
+        return {
+            "create_model_from_job": ("sagemaker_session", "sess"),
+        }
+
+    @property
+    def old_param_name(self):
+        """The previous name for the image URI argument."""
+        return "primary_container_image"
+
+    @property
+    def new_param_name(self):
+        """The new name for the the image URI argument."""
+        return "image_uri"
+
+
+class SessionCreateEndpointImageURIRenamer(MethodParamRenamer):
+    """A class to rename ``deployment_image`` to ``image_uri``.
+
+    This looks for the following calls:
+
+    - ``sagemaker_session.endpoint_from_job()``
+    - ``sess.endpoint_from_job()``
+    - ``sagemaker_session.endpoint_from_model_data()``
+    - ``sess.endpoint_from_model_data()``
+    """
+
+    @property
+    def calls_to_modify(self):
+        """A mapping of the ``endpoint_from_*`` functions to common variable names for Session."""
+        return {
+            "endpoint_from_job": ("sagemaker_session", "sess"),
+            "endpoint_from_model_data": ("sagemaker_session", "sess"),
+        }
+
+    @property
+    def old_param_name(self):
+        """The previous name for the image URI argument."""
+        return "deployment_image"
+
+    @property
+    def new_param_name(self):
+        """The new name for the the image URI argument."""
+        return "image_uri"
```

Comment:
fft: although i don't like promoting deep class inheritance hierarchy, this method is common to all `*ImageURIRenamer` and may benefit from some inheritance

Reply:
(talked offline) I'd also noticed the duplication too, but agree that too many layers of inheritance also isn't great